app/src/main/java/org/openobservatory/ooniprobe/activity/TextActivity.java

```diff
@@ -4,31 +4,85 @@
 import android.content.ClipboardManager;
 import android.content.Context;
 import android.content.Intent;
+import android.graphics.Bitmap;
 import android.os.Bundle;
 import android.view.Menu;
 import android.view.MenuItem;
-import android.widget.TextView;
+import android.view.View;
+import android.webkit.WebChromeClient;
+import android.webkit.WebResourceError;
+import android.webkit.WebResourceRequest;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+import android.widget.ProgressBar;
 import android.widget.Toast;
 
+import org.apache.commons.io.FileUtils;
 import org.openobservatory.ooniprobe.R;
+import org.openobservatory.ooniprobe.client.callback.GetMeasurementJsonCallback;
+import org.openobservatory.ooniprobe.client.callback.GetMeasurementsCallback;
+import org.openobservatory.ooniprobe.common.ReachabilityManager;
+import org.openobservatory.ooniprobe.model.api.ApiMeasurement;
+import org.openobservatory.ooniprobe.model.database.Measurement;
 
 import androidx.annotation.Nullable;
+
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParser;
+
+import java.io.File;
+import java.nio.charset.Charset;
+
 import butterknife.BindView;
 import butterknife.ButterKnife;
+import localhost.toolkit.app.fragment.MessageDialogFragment;
+import okhttp3.Request;
 
 public class TextActivity extends AbstractActivity {
-	private static final String TEXT = "text";
-	@BindView(R.id.textView) TextView textView;
+	private Measurement measurement;
+	private String text;
+	private static final int TYPE_LOG = 1;
+	private static final int TYPE_JSON = 2;
+	private static final String TEST = "test";
+	private static final String TYPE = "type";
+	@BindView(R.id.webView) WebView webView;
+	@BindView(R.id.progressBar) ProgressBar progressBar;
 
-	public static Intent newIntent(Context context, String text) {
-		return new Intent(context, TextActivity.class).putExtra(TEXT, text);
+	public static Intent newIntent(Context context, int type, Measurement measurement) {
+		return new Intent(context, TextActivity.class).putExtra(TYPE, type).putExtra(TEST, measurement);
 	}
 
 	@Override protected void onCreate(@Nullable Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 		setContentView(R.layout.text);
 		ButterKnife.bind(this);
-		textView.setText(getIntent().getStringExtra(TEXT));
+		measurement = (Measurement) getIntent().getSerializableExtra(TEST);
+		webView.setWebChromeClient(new WebChromeClient());
+		setProgressBarVisibility(View.VISIBLE);
+		webView.setWebViewClient(new WebViewClient() {
+			@Override
+			public void onPageStarted(WebView view, String url, Bitmap favicon) {
+				super.onPageStarted(view, url, favicon);
+				setProgressBarVisibility(View.VISIBLE);
```

Comment:
Isn't this made redundant by a similar call above?

Reply:
yeah makes sense